src/mmda/parsers/grobid_augment_existing_document_parser.py

```diff
@@ -61,61 +61,87 @@ def parse(self, input_pdf_path: str, doc: Document, xml_out_dir: Optional[str] =
+        annotatable_span_groups_by_field_name = self._grobid_box_groups_to_span_groups(doc, **box_group_parses)
+        doc.annotate(**annotatable_span_groups_by_field_name)
 
         return doc
 
-    def _xml_coords_to_boxes(self, coords_attribute: str, page_sizes: dict):
+    def _xml_coords_to_boxes(self, coords_attribute: str):
         coords_list = coords_attribute.split(";")
         boxes = []
         for coords in coords_list:
             pg, x, y, w, h = coords.split(",")
             proper_page = int(pg) - 1
             boxes.append(
                 Box(
-                    l=float(x), 
-                    t=float(y), 
-                    w=float(w), 
-                    h=float(h), 
+                    l=float(x),
+                    t=float(y),
+                    w=float(w),
+                    h=float(h),
                     page=proper_page
-                ).get_relative(*page_sizes[proper_page])
+                ).get_relative(*self.page_sizes[proper_page])
             )
         return boxes
 
-    def _get_grobid_bib_box_groups(self, root: et.Element) -> List[BoxGroup]:
-        bib_list_root = root.find(".//tei:listBibl", NS)
-
+    def _cache_page_sizes(self, root: et.Element):
         page_size_root = root.find(".//tei:facsimile", NS)
         page_size_data = page_size_root.findall(".//tei:surface", NS)
         page_sizes = dict()
         for data in page_size_data:
             page_sizes[int(data.attrib["n"]) - 1] = [float(data.attrib["lrx"]), float(data.attrib["lry"])]
-        
-        grobid_bibs = []
+        self.page_sizes = page_sizes
+        return page_sizes
+
+    def _get_author_box_groups(self, root: et.Element) -> List[BoxGroup]:
+        author_list_root = root.find(".//tei:sourceDesc", NS)
+
+        author_names = []
+        author_name_structs = author_list_root.findall(".//tei:persName", NS)
+        for a in author_name_structs:
+            coords_string = a.attrib["coords"]
+            boxes = self._xml_coords_to_boxes(coords_string)
+
+            author_names.append(
+                BoxGroup(boxes=boxes)
+            )
+
+        return author_names
+
+    def _get_bib_entry_box_groups(self, root: et.Element) -> List[BoxGroup]:
+        bib_list_root = root.find(".//tei:listBibl", NS)
+
+        bib_entries = []
         bib_structs = bib_list_root.findall(".//tei:biblStruct", NS)
         for bib in bib_structs:
             coords_string = bib.attrib["coords"]
-            boxes = self._xml_coords_to_boxes(coords_string, page_sizes)
+            boxes = self._xml_coords_to_boxes(coords_string)
             grobid_id = bib.attrib["{http://www.w3.org/XML/1998/namespace}id"]
 
-            grobid_bibs.append(
+            bib_entries.append(
                 BoxGroup(
-                boxes=boxes, 
-                metadata=Metadata(grobid_id=grobid_id)
+                    boxes=boxes,
+                    metadata=Metadata(grobid_id=grobid_id)
```

Comment:
It feels like this is the only reason we can't write one `_get_grobid_box_groups` method which just gets passed the needed tei keys for a given element. Do we definitely need this?

Reply:
That's a good question ... we don't definitely need this, unless we certainly want to make Grobid mentions and links to bibs available from SPP. If it turns out we do need this, i think there can be ways to add in optional extra metadata from a single `_get_grobid_box_groups` method -- and for now since we aren't even using that id I'm happy to take it out and try refactoring a little